doc/source/whatsnew/v0.19.0.txt

```diff
@@ -854,4 +854,9 @@ Bug Fixes
 - Bugs in ``Index.difference`` and ``DataFrame.join`` raise in Python3 when using mixed-integer indexes (:issue:`13432`, :issue:`12814`)
 
 - Bug in ``.to_excel()`` when DataFrame contains a MultiIndex which contains a label with a NaN value (:issue:`13511`)
+
 - Bug in ``pd.read_csv`` in Python 2.x with non-UTF8 encoded, multi-character separated data (:issue:`3404`)
+
+- Bug in ``agg()`` function on groupby dataframe changes dtype of ``datetime64[ns]`` column to ``float64`` (:issue:`12821`)
+
+- Bug in operations on ``NaT`` returning ``float`` instead of ``datetime64[ns]`` (:issue:`12941`)
```

Comment:
Both #12941 and #12821 are fixed in the PR. The two bugs behave similarly, but NaT is handled in different locations when casted. 

Please see commits in the PR:
+ #12821 is fixed in commit [ BUG: fix GH12821, agg() function on groupby dataframe changes dtype o… ](https://github.com/pydata/pandas/pull/12992/commits/a949cee9d2aac812c4155c174a0360780b0cce39)

+ #12941 is fixed in commit [BUG: fix GH12941, Operations on NaT returning float instead of dateti…](https://github.com/pydata/pandas/pull/12992/commits/8d17eedf5db53d4d1f8ad7fbc77b329902b1610f)